execution_engine/src/runtime_context/mod.rs

```diff
@@ -402,8 +437,13 @@ where
     ///
     /// This could be either a [`Key::Account`] or a [`Key::Hash`] depending on the entry point
     /// type.
-    pub fn base_key(&self) -> Key {
-        self.base_key
+    pub fn get_entity_address(&self) -> Key {
+        self.entity_address
+    }
+
+    /// Returns the initiater of the call chain.
+    pub fn get_caller(&self) -> AccountHash {
+        self.account_hash
     }
```

Comment:
maybe (AccountHash, AddressableEntityHash) or a newtype of same will be more ergonomic